### PR TITLE
Add Default Reg Status to Event GQL Data Type and Mutator

### PR DESCRIPTION
**Pull request by tn3rb**
Noticed while reviewing https://github.com/eventespresso/barista/pull/514 that out GQL Event object was missing a value for the default registration status.

This PR adds that in so that it can be used to wire up that input in the new Reg Options "meta box" in the EDTR

### CODE REVIEW

File: core/domain/services/graphql/data/mutations/EventMutation.php [hunk 1]
```diff
@@ -28,35 +28,39 @@ public static function prepareFields(array $input, $mutation_name)
         $args = [];
 
         if (array_key_exists('allowDonations', $input)) {
-            $args['EVT_donations'] = (bool) ($input['allowDonations']);
+            $args['EVT_donations'] = filter_var($input['allowDonations'], FILTER_VALIDATE_BOOLEAN);
```
Comment (manzoorwanijk) [inline core/domain/services/graphql/data/mutations/EventMutation.php:31]: Nice 👍

File: core/domain/services/graphql/types/Event.php [hunk 2]
```diff
@@ -79,22 +73,22 @@ public function getFields()
...
+                'dbId',
+                ['non_null' => 'Int'],
+                'ID',
+                esc_html__('The event ID.', 'event_espresso')
             ),
-            new GraphQLInputField(
-                'manager',
+            new GraphQLField(
+                'defaultRegStatus',
                 'String',
```
Comment (manzoorwanijk) [inline core/domain/services/graphql/types/Event.php:83]: IMHO, we should allow a limited set of statuses by using enum instead of string.

**Review by manzoorwanijk — COMMENTED** [review]
Just have that one suggestion.

File: core/domain/services/graphql/types/Event.php (hunk 2 above)
Comment (tn3rb) [reply core/domain/services/graphql/types/Event.php:83]: great suggestion... will implement, thnx

File: core/domain/services/graphql/types/Event.php [hunk 3]
```diff
@@ -79,22 +72,22 @@ public function getFields()
...
-            new GraphQLInputField(
-                'manager',
-                'String',
-                null,
-                esc_html__('Globally unique event ID for the event manager', 'event_espresso')
+            new GraphQLOutputField(
+                'defaultRegStatus',
+                $this->namespace . 'RegistrationStatusEnum',
+                'default_registration_status',
+                esc_html__('Default Event Registration Status', 'event_espresso')
```
Comment (tn3rb) [inline core/domain/services/graphql/types/Event.php:84]: this is the important change in this PR

**Review by manzoorwanijk — APPROVED** [review]
Needed a minor fix. Looks a great

**Pull request merged**